minidump-unwind/src/symbols/debuginfo.rs

```diff
@@ -424,6 +449,39 @@ fn load_unwind_module(module: &dyn Module) -> Option<(Mmap, framehop::Module<Mod
     Some((mapped, fhmodule))
 }
 
+fn get_fat_macho_data<'data, Fat: FatArch>(
+    path: &Path,
+    fatfile_data: &'data [u8],
+    result: object::read::Result<object::read::macho::MachOFatFile<'data, Fat>>,
+    arch: Architecture,
+) -> Option<&'data [u8]> {
+    match result {
+        Err(e) => {
+            tracing::error!("failed to parse fat macho file {}: {e}", path.display());
+            None
+        }
+        Ok(fatfile) => {
+            let Some(arch) = fatfile.arches().iter().find(|a| a.architecture() == arch) else {
+                tracing::error!(
+                    "failed to find object file for {arch:?} architecture in fat macho file {}",
+                    path.display()
+                );
+                return None;
+            };
+            match arch.data(fatfile_data) {
+                Err(e) => {
+                    tracing::error!(
+                        "failed to read data from fat macho file {}: {e}",
+                        path.display()
+                    );
+                    None
+                }
+                Ok(data) => Some(data),
+            }
```

Comment:
You can probable use `map_or_else()` here.

Reply:
That's one I've needed to reach for infrequently enough that I forget about it :smile: